packages/adapter-components/src/references/reference_mapping.ts

```diff
@@ -70,6 +70,27 @@ export const ReferenceSerializationStrategyLookup: Record<
   },
 }
 
+export type ReferenceValidationStrategy = {
+  validate: (referringFieldValue: string|number, serializedRefExpr: string) => boolean
+}
```

Comment:
do we anticipate other extensions of this type? maybe we don't need the `validate` function name and can just have this be
```suggestion
export type ReferenceValidationStrategy = (referringFieldValue: string|number, serializedRefExpr: string) => boolean
```
with the customizations below being like
```
  asString: (referringFieldValue, serializedRefExpr) => (
    _.toString(referringFieldValue) === _.toString(serializedRefExpr)
  ),
```
?
the reason we had function names in the serialization part was that we had two mirroring functions that had to be combined, but here I don't think we need this yet (and if/when we do, I'm not convinced it will be under the same strategy and not customizable per rule)